R/aaa_survival_prop.R

```diff
@@ -185,21 +189,36 @@ survival_prob_coxnet <- function(object, new_data, times, output = "surv", ...)
     object$fit,
     newx = new_x,
     newstrata = new_strata,
+    s = penalty,
     x = object$training_data$x,
     y = object$training_data$y,
     na.action = na.exclude,
     ...
   )
 
-  res <- stack_survfit(y, nrow(new_data)) %>%
-    dplyr::group_nest(.row, .key = ".pred") %>%
+  if (multi) {
+    names(y) <- penalty
+    keep_penalty <- TRUE
+    stacked_survfit <-
+      purrr::map_dfr(y, stack_survfit, n = nrow(new_data), .id = "penalty") %>%
+      dplyr::mutate(penalty = as.numeric(penalty)) %>%
```

Comment:
I like that idea but it looks like there may be a bug when using it like this :eyes: 
``` r
library(tibble)

# this is fine
tibble(x = 1:3, y = NULL)
#> # A tibble: 3 x 1
#>       x
#>   <int>
#> 1     1
#> 2     2
#> 3     3

# but not in a function
make_tibble <- function(x = 1:3, y = NULL) {
  tibble(x = x,
         y = y)
}
make_tibble(y = NULL)
#> Error: Internal error in `df_slice()`: Columns must match the data frame size.

# the function is okay for a non-NULL y
make_tibble(y = 1:3)
#> # A tibble: 3 x 2
#>       x     y
#>   <int> <int>
#> 1     1     1
#> 2     2     2
#> 3     3     3
```

<sup>Created on 2021-07-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

Reply:
update: bug report by @DavisVaughan https://github.com/tidyverse/tibble/issues/900

Reply:
turned it into #71 to solve once tibble gets updated

Reply:
Why is `penalty` a character string? That seems problematic.

Reply:
I use it to name the survfit object so that `map_dfr()` adds the correct penalty to each tibble resulting from `stack_survfit()`. Using it as a name is what turns it into a character. I'll change that in the way that Davis suggested once the bug in tibble is fixed :broom: